crates/oxc_linter/src/rules/deepscan/bad_array_method_on_arguments.rs

```diff
@@ -0,0 +1,197 @@
+    /// Otherwise, a TypeError exception will be thrown because of the non-existent method.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// function add(x, y) {
+    ///   return x + y;
+    /// }
+    /// function sum() {
+    ///   return arguments.reduce(add, 0);
+    /// }
+    /// ```
+    BadArrayMethodOnArguments,
+    correctness,
+);
+
+impl Rule for BadArrayMethodOnArguments {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::IdentifierReference(reference) = node.get().kind() 
+            && reference.name == "arguments"
+            && let Some(parent_node) = ctx.parent_node(node)
+            && let AstKind::MemberExpression(member_expr) = parent_node.get().kind()
+            && let Some(parent_node) = ctx.parent_node(parent_node)
+            && let AstKind::CallExpression(_) = parent_node.get().kind()
+        {
+            match member_expr {
+                MemberExpression::StaticMemberExpression(expr) => {
+                    if ARRAY_METHODS.binary_search(&expr.property.name.as_str()).is_ok() {
+                        ctx.diagnostic(BadArrayMethodOnArgumentsDiagnostic(
+                            expr.property.name.clone(),
+                            expr.span,
+                        ));
+                    }
+                }
+                MemberExpression::ComputedMemberExpression(expr) => {
+                    match &expr.expression {
+                        Expression::StringLiteral(name) => {
+                            if ARRAY_METHODS.binary_search(&name.value.as_str()).is_ok() {
+                                ctx.diagnostic(BadArrayMethodOnArgumentsDiagnostic(
+                                    name.value.clone(),
+                                    expr.span,
+                                ));
+                            }
+                        }
+                        Expression::TemplateLiteral(template) => {
+                            // only check template string like "arguments[`METHOD_NAME`]" for Deepscan compatible
+                            if template.expressions.is_empty() 
+                                && template.quasis.len() == 1
+                                && let Some(template_element) = template.quasis.get(0)
+                                && let Some(name) = template_element.value.cooked.as_deref()
+                                && ARRAY_METHODS.binary_search(&name).is_ok()
+                            {
+                                ctx.diagnostic(BadArrayMethodOnArgumentsDiagnostic(
+                                    Atom::new(name), 
+                                    expr.span,
+                                ));
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+                MemberExpression::PrivateFieldExpression(_) => unreachable!()
```

Comment:
Empty here is fine, I don't think you can reach here for code coverage, you don't need to aim for 100%.